arm64/math-vector.h: allow disabling vector types via __ARM_VEC_MATH_DISABLED

Minimal rework of Simon Chopin's RFC (BZ #25422) for glibc 2.44: C parsers
using GCC as preprocessor (FEX thunk generation) fail on the SIMD intrinsic
typedefs; -D__ARM_VEC_MATH_DISABLED=1 bypasses them.

--- a/sysdeps/aarch64/fpu/bits/math-vector.h
+++ b/sysdeps/aarch64/fpu/bits/math-vector.h
@@ -191,17 +191,18 @@
 # define __DECL_SIMD_tanpif __DECL_SIMD_aarch64
 #endif
 
-#if __GNUC_PREREQ(9, 0)
+#if !defined __ARM_VEC_MATH_DISABLED && __GNUC_PREREQ(9, 0)
 #  define __ADVSIMD_VEC_MATH_SUPPORTED
 typedef __Float32x4_t __f32x4_t;
 typedef __Float64x2_t __f64x2_t;
-#elif __glibc_clang_prereq(8, 0)
+#elif !defined __ARM_VEC_MATH_DISABLED && __glibc_clang_prereq(8, 0)
 #  define __ADVSIMD_VEC_MATH_SUPPORTED
 typedef __attribute__ ((__neon_vector_type__ (4))) float __f32x4_t;
 typedef __attribute__ ((__neon_vector_type__ (2))) double __f64x2_t;
 #endif
 
-#if __GNUC_PREREQ(10, 0) || __glibc_clang_prereq(11, 0)
+#if !defined __ARM_VEC_MATH_DISABLED \
+    && (__GNUC_PREREQ(10, 0) || __glibc_clang_prereq(11, 0))
 #  define __SVE_VEC_MATH_SUPPORTED
 typedef __SVFloat32_t __sv_f32_t;
 typedef __SVFloat64_t __sv_f64_t;
